### ISSUE TITLE
Clars "to all teams" don't show up in WTI

### ISSUE THREAD

**Issue by clevengr** [pc2ccs/pc2v9#186]
**Describe the issue**: 

As reported by Tom Rokicki in an email 2/26/21 4:04pm:

I just tested sending a clarification to all.

The response was received only by the submitting team.  No other team got any
popup announcing a new clar response, nor any entry in the Clarifications list on
their web interface (even when they logged out and back in).

This is a pretty high severity issue.

**Comment by lane55** [comment]
### Background and root cause

This bug only appears when there are teams assigned different groups/divisions.
A team in WTI and I would guess in a pc2 team module will not be shown broadcast clars from
other groups.    

When the earlier bug was discovered it caused NO clars to be shown to 
the team if they received a broadcast clar from a different division.   The fix
for that earlier bug was to show the clars for the team's (group) and do
not display broadcast clars, rather than show no clars to the team.
The root cause still existed and exists.  It has to do with the Problems
loaded into the teams model.

When a team is a division, the team is shown only its groups problems,
this was done by only loading the team's assigned problems into the its model.
The earlier bug caused an Exception which, in effect, did not show any clars
to the team.   That bug was fixed but if a broadcast clar has a problem that
is not assigned to the team - when the team API fetches the problem title
from the model - the API cannot find that Problem and simply does not display
the broadcast clar.

If a broadcast clar is from the same group that is assigned to the team, then
the clar will be shown to the team.

It is the cross division broadcast clars that are not displayed.

**Comment by lane55** [comment]
Marked as high priority because as stated this does not work.
Broadcast clars are critical to a contest and communication with teams.

**Comment by clevengr** [comment]
After extensive testing, I'm unable to duplicate this problem.  I think it may have been inadvertently fixed in the changes that were made around Feb/Mar 2021.

Specifically, I followed the (updated) "Steps to Reproduce" as listed in the Issue:
- I loaded a contest with 5 problems and four groups (the "ccs1" sample contest).
- I assigned the first problem ("APL") to (just) the "North" group.
- I assigned the second problem ("BarCodes") to (just) the "South" group.
- I logged in via WTI to "team1", which is in the North group.
- I verified that team 1 could see the "APL" problem, and could NOT see the "BarCodes" problem.
- I logged in (via WTI) using a separate browser window as "team41", which is in the "South" group.
- I verified that team 41 could see the "BarCodes" problem and could NOT see the "APL" problem.
- I submitted a clar from team 1 for the APL problem.
- I answered the clar using a manual judge, and I checked "Send to All Teams".
- Team 1 got the clarification answer.
- **_Team 41 also got the clar answer_**, even though the "APL" problem was not listed in their problem list on the Submission screen.

Note:  the "problem title" which shows up on the Team 41 clar screen is NOT the same as the problem title which shows up on the Team 1 clar screen.  
Here is the Team 1 screen showing the clar response:

![image](https://github.com/pc2ccs/pc2v9/assets/9256691/901c9eaf-4e9f-4276-bf33-71269a0e3ed5)

Note that Team 1 can see the APL problem (along with three others), but not the BarCodes problem, and the problem title in the clar response is "APL".

Here is the Team 41 screen showing the clar response (which according to the Issue should not have been received):

![image](https://github.com/pc2ccs/pc2v9/assets/9256691/a2fe5e48-ff5f-40b1-9542-70326a72500d)

Note that Team 41 CANNOT see the "APL" problem in their drop-down problem list, but they DID get the "Send to All Teams" clar response.  Note also that the problem title in the clar response is "apl-", not "APL".  ("APL" is the problem _name_; "apl" is the problem _short-name_). 

Bottom line:  clar responses marked "Send to All Teams" DO show up in teams which can't see the problem.
Unless someone else can duplicate the problem defined by this Issue, I'm prepared to mark this issue "Closed, Cannot reproduce".

(Additional note:  there are two related issues, having to do with lack of a "General Clar" selection:  Issue #448 and Issue #180.  However, I don't think those have direct bearing on this issue.)

**Comment by johnbrvc** [comment]
I saw no problem as described above.  I followed the procedure above AND also on the remnants of a copy of the Greater NY Regional.  Both teams received the clarification answer.  Of course, as previously discussed, this begs the question SHOULD the 2nd team receive the clarification since it was for a problem they do not see?  

If a judge wants to send a clarification to EVERYONE (general), they should generate a general clar and send it out as opposed to answering a question about a problem that some teams can't see or care about.  If the response is general enough, it should be a general clar, not a problem specific clar.  But, that is another issue.
